fvm/evm/types/address.go

```diff
@@ -61,6 +66,23 @@ func NewAddressFromBytes(inp []byte) Address {
 	return Address(gethCommon.BytesToAddress(inp))
 }
 
+func COAAddressFromFlowEvent(evmContractAddress flow.Address, event flow.Event) (Address, error) {
+	// check the type first
+	if string(event.Type) != fmt.Sprintf(COAAddressTemplate, evmContractAddress.Hex()) {
```

Comment:
Maybe optimize this check by avoiding the string formatting/allocation, and just decode the event's type ID and comparing address and qualified identifier

Reply:
This is not a code that is touched by the onchain execution but I'll update the next PR to do as you suggested. How do I decode the event type ID? is there a utility function on the cadence side to decode from string?